### ISSUE TITLE
Error when restarting `guard` from bin/guard

### ISSUE THREAD

**Issue by deivid-rodriguez** [guard/guard#727]
Hi, first of all, thanks for your awesome late work in guard and its plugins! :+1: 

So I edited my Guardfile and `guard` exited and pointed me [here](https://github.com/guard/guard/wiki/Guard-2.10.3-exits-when-Guardfile-is-changed). I read it and decided to edit `bin/guard` and go for the alternative workaround. This is want I wrote

    #!/usr/bin/env ruby
    #
    # The application 'guard' is installed as part of a gem, and
    # this file is here to facilitate running it.
    #
    require 'pathname'
    ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile',
                                               Pathname.new(__FILE__).realpath)

    require 'rubygems'
    require 'bundler/setup'

    begin
      load Gem.bin_path('guard', 'guard')
    rescue SystemExit => e
      exit(e.status) unless e.status == 2
      STDOUT.puts 'Restarting guard...'
      retry
    end

And I get this error message I'm getting

    21:39:37 - INFO - Bye bye...
    Restarting guard...
    /home/davidr/.rvm/gems/ruby-2.2.0/gems/lumberjack-1.0.9/lib/lumberjack/device/writer.rb:58:in `sync=': closed stream (IOError)
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/lumberjack-1.0.9/lib/lumberjack/device/writer.rb:58:in `initialize'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/lumberjack-1.0.9/lib/lumberjack/logger.rb:281:in `new'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/lumberjack-1.0.9/lib/lumberjack/logger.rb:281:in `open_device'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/lumberjack-1.0.9/lib/lumberjack/logger.rb:66:in `initialize'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/guard-2.11.1/lib/guard/ui.rb:27:in `new'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/guard-2.11.1/lib/guard/ui.rb:27:in `logger'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/guard-2.11.1/lib/guard/aruba_adapter.rb:49:in `ensure in execute'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/guard-2.11.1/lib/guard/aruba_adapter.rb:56:in `execute'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/guard-2.11.1/lib/guard/aruba_adapter.rb:19:in `execute!'
    from /home/davidr/.rvm/gems/ruby-2.2.0/gems/guard-2.11.1/bin/guard:11:in `<top (required)>'
    from bin/guard:14:in `load'
    from bin/guard:14:in `<main>'

Not sure whether the problem lies in my script, in guard or in lumberjack, but I thought it was worth reporting it.

Thanks!!

**Comment by e2** [comment]
Thanks - I really appreciate the support! Thanks for your work as well (it's the hours spent that no one ever know about that count the most).

To be honest, I haven't worked this one out yet (I took a stab at it, but gave up - sorry, I at least should have left some "breadcrumbs" to where I got..).

The quickfix is to just call guard within a subprocess, e.g.

```ruby
system(Gem.bin_path('guard', 'guard'))
```
and then check `$?`.

The cause of the problem is that the aruba_wrapper closes the stderr stream when the guard cli is finished - and the logger can't reopen it. I would probably have to add an option not to close the logger.

Either way, I should *at least* provide a working implementation of `bin/guard`.

Thanks for letting me know!
